### PR TITLE
feat(seo): add restoreable meta service

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #1732 

#1704 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/seo/src/meta/get-attr-selector.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DaffSeoMetaDefinition } from '../models/public_api';
+
+export function getAttrSelector(def: DaffSeoMetaDefinition): string {
+  // TODO: escape value?
```
Comment (damienwebdev) [inline libs/seo/src/meta/get-attr-selector.ts:4]: We don't need this to be escaped.

File: libs/seo/src/meta/get-attr-selector.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { DaffSeoMetaDefinition } from '../models/public_api';
+
+export function getAttrSelector(def: DaffSeoMetaDefinition): string {
```
Comment (damienwebdev) [inline libs/seo/src/meta/get-attr-selector.ts:3]: Can you add a doc?

File: libs/seo/src/meta/meta.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { DaffSeoMetaDefinition } from '../models/public_api';
+import { getAttrSelector } from './get-attr-selector';
+
+interface Subtype {
+  name: string;
+  content: string;
+}
+
+/**
+ * @inheritdoc
```
Comment (damienwebdev) [inline libs/seo/src/meta/meta.service.ts:16]: This needs a doc.

File: libs/seo/src/meta/meta.service.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class DaffMetaService {
+  constructor(
+    private meta: Meta,
+  ) {}
+
+  upsert(def: DaffSeoMetaDefinition): void {
```
Comment (damienwebdev) [inline libs/seo/src/meta/meta.service.ts:26]: Can you add method comments?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/seo/src/meta/get-attr-selector.ts (hunk 1 above)
Comment (griest024) [reply libs/seo/src/meta/get-attr-selector.ts]: removed todo

File: libs/seo/src/meta/get-attr-selector.ts (hunk 2 above)
Comment (griest024) [reply libs/seo/src/meta/get-attr-selector.ts:8]: added

File: libs/seo/src/meta/meta.service.ts (hunk 3 above)
Comment (griest024) [reply libs/seo/src/meta/meta.service.ts]: added

File: libs/seo/src/meta/meta.service.ts (hunk 4 above)
Comment (griest024) [reply libs/seo/src/meta/meta.service.ts:23]: added

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**